tests/test_cnes_sonarqube.py

```diff
@@ -0,0 +1,251 @@
+
+    def test_eus_admin(self):
+        """
+        Evil User Story:
+        As a hacker, I want to use the default admin password ("admin")
+        to log in as admin.
+        """
+        status = requests.post(f"{self.SONARQUBE_URL}/api/authentication/login",
+                    auth=("admin", "admin"),
+                    data={"login": "admin", "password": "admin"}).status_code
+        # Hint: if this test fails, a hacker may have logged in (HTTP 200) or crash the server (HTTP 500+)
+        assert status == 401 # Unauthorized
+
+    def test_no_config_twice(self):
+        """
+        As a SonarQube user, I want the configuration of the server
+        not to be executed if the server has already been configured
+        so that the database is not populated more than once.
+        """
+        # On Linux, max_map_count must be at least 262144 to run this test
+        if platform.system() == 'Linux':
+            proc_stdout = subprocess.run("sysctl -a", shell=True, check=True, capture_output=True).stdout
+            max_map_count = int(re.findall('vm.max_map_count = [0-9]+', str(proc_stdout))[0].split(' = ')[1])
+            # Hint: if this test fails, run: sudo sysctl -w vm.max_map_count=262144
+            assert max_map_count >= 262144
+        docker_client = docker.from_env()
+        lequalsonarqube_container_name="lequalsonarqube-compose"
+        # Use the compose file with an external database
+        print("Starting the service (sonarqube and postgres)...")
+        subprocess.run("docker-compose up -d", shell=True, check=True, capture_output=True)
+        # Wait for the SonarQube container to be configured
+        self.wait_cnes_sonarqube_ready(lequalsonarqube_container_name)
+        # Restart the SonarQube server but not the database
+        print("Restarting SonarQube server...")
+        docker_client.containers.get(lequalsonarqube_container_name).restart()
+        time.sleep(30)
+        self.wait_cnes_sonarqube_ready(lequalsonarqube_container_name, tail=10)
+        # Check SonarQube logs
+        config_logs = docker_client.containers.get(lequalsonarqube_container_name).logs()
+        subprocess.run("docker-compose down", shell=True, check=True, capture_output=True)
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_data").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_extensions").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_logs").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql_data").remove()
+        # Hint: if this test fails, the server may have been reconfigured when not needed or the test did not wait long enough for the expected message to be logged
+        assert b"[INFO] CNES SonarQube: The database has already been filled with CNES configuration. Not adding anything." in config_logs
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)